app/src/main/java/org/schabi/newpipe/MainActivity.java

```diff
@@ -167,6 +169,10 @@ protected void onCreate(final Bundle savedInstanceState) {
             // if this is enabled by the user.
             NotificationWorker.initialize(this);
         }
+        if (ReleaseVersionUtil.INSTANCE.isReleaseApk()
+                && !UpdateSettingsFragment.wasUserAskedForConsent(this)) {
```

Comment:
Isn't running the checks in the contrary order better? I think for release builds, in the case when consent has been already asked, checking for consent first has a lower cost than searching for signatures.